src/main/java/ch/njol/skript/effects/EffWorldUnload.java

```diff
@@ -0,0 +1,74 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Unload World")
+@Description({"Unload a world"})
+@Examples({
+	"unload \"world_nether\" and save",
+	"unload \"world_the_end\" and don't save",
+	"unload all worlds"
+})
+@Since("INSERT VERSION")
+public class EffWorldUnload extends Effect {
+
+	static {
+		Skript.registerEffect(EffWorldUnload.class, "unload %worlds% [and (:save)]");
+	}
+
+	private boolean save;
+	private Expression<World> worlds;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		worlds = (Expression<World>) exprs[0];
+		save = parseResult.hasTag("save");
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		World mainWorld = Bukkit.getWorlds().get(0);
+		for (World world : this.worlds.getArray(event)) {
+			if (mainWorld != null && world != mainWorld)
+				world.getPlayers().forEach(player -> player.teleport(mainWorld.getSpawnLocation()));
+			Bukkit.unloadWorld(world, save);
```

Comment:
What happens if you unload a world containing players?

Reply:
I remember originally having a talk somewhere about this, if players are in the world 1 of 2 things happen. they are teleported to the main world or it just fails to unload

since I remember testing this in the `mainWorld` and it just didn't do anything didn't teleport me to another world or unload the world, so I think I did this as a backup safe check

Reply:
For above, I've revisited this and it seems to be working fine without the teleport portion, unloads `mainWorld` teleports me to one of the others, unload that world resends me to the `mainWorld`. Seems like `mainWorld` still can't be unloaded but will atleast teleport players outside of it